build.zig

```diff
@@ -220,225 +227,157 @@ pub fn getWrapperModuleVulkan(sdk: *Sdk, vulkan: *Build.Module) *Build.Module {
+        .target = exe.root_module.resolved_target.?,
+        .optimize = exe.root_module.optimize.?,
+    });
+    build_linux_sdl_stub.addAssemblyFile(sdk.prepare_sources.getStubFile());
+    exe.linkLibrary(build_linux_sdl_stub);
+}
 
-    if (target.result.os.tag == .linux) {
-        if (!is_native) {
-            @panic("Cannot cross-compile with TTF to linux yet.");
+fn linkWindows(
+    sdk: *Sdk,
+    exe: *Compile,
+    linkage: std.builtin.LinkMode,
+    comptime library: Library,
+    paths: Paths,
+) !void {
+    exe.addIncludePath(.{ .cwd_relative = paths.include });
+    exe.addLibraryPath(.{ .cwd_relative = paths.libs });
+
+    const lib_name = switch (library) {
+        .SDL2 => "SDL2",
+        .SDL2_ttf => "SDL2_ttf",
+    };
+
+    if (exe.root_module.resolved_target.?.result.abi == .msvc) {
+        exe.linkSystemLibrary2(lib_name, .{ .use_pkg_config = .no });
+    } else {
+        const file_name = try std.fmt.allocPrint(sdk.build.allocator, "lib{s}.{s}", .{
+            lib_name,
+            if (linkage == .static) "a" else "dll.a",
+        });
+        defer sdk.build.allocator.free(file_name);
+
+        const lib_path = try std.fs.path.join(sdk.build.allocator, &[_][]const u8{ paths.libs, file_name });
+        defer sdk.build.allocator.free(lib_path);
+
+        exe.addObjectFile(.{ .cwd_relative = lib_path });
+
+        if (linkage == .static and library == .SDL2) {
+            const static_libs = [_][]const u8{
+                "setupapi",
+                "user32",
+                "gdi32",
+                "winmm",
+                "imm32",
+                "ole32",
+                "oleaut32",
+                "shell32",
+                "version",
+                "uuid",
+            };
+            for (static_libs) |lib| exe.linkSystemLibrary(lib);
         }
+    }
 
-        // on linux with compilation for native target,
-        // we should rely on the system libraries to "just work"
-        exe.linkSystemLibrary("sdl2_ttf");
-    } else if (target.result.os.tag == .windows) {
-        @panic("Cannot link TTF on windows yet.");
-    } else if (target.result.isDarwin()) {
-        if (!host_system.os.tag.isDarwin())
-            @panic("Cannot cross-compile with TTF to macOS yet.");
+    if (linkage == .dynamic and exe.kind == .exe) {
+        const dll_name = try std.fmt.allocPrint(sdk.build.allocator, "{s}.dll", .{lib_name});
+        defer sdk.build.allocator.free(dll_name);
+
+        const dll_path = try std.fs.path.join(sdk.build.allocator, &[_][]const u8{ paths.bin, dll_name });
+        defer sdk.build.allocator.free(dll_path);
+
+        sdk.build.installBinFile(dll_path, dll_name);
+    }
+}
+
+fn linkMacOS(exe: *Compile, comptime library: Library) !void {
+    exe.linkSystemLibrary(switch (library) {
+        .SDL2 => "sdl2",
+        .SDL2_ttf => "sdl2_ttf",
+    });
 
-        exe.linkSystemLibrary("sdl2_ttf");
+    if (library == .SDL2) {
+        exe.linkFramework("IOKit");
```

Comment:
The `for (static_libs) |lib| ...` is used above. And I guess we could also do the same for all these, but I saw no real benefit in doing so with these, so I left these as-is. (Having these in a list and iterating over the list wouldn't even reduce the line count)